### ISSUE TITLE
[clang-tidy] false positive for modernize-type-traits with structured bindings

### ISSUE THREAD

**Issue by matthew-f** [llvm/llvm-project#153649]
Hi 

Using an llvm update from a few days `modernize-type-traits` is generating what looks like a false positive. An update from today still exhibits the same problem
```
std::map<int, int> map = {{1, 2}, {3, 4}};

for (const auto& [a, b] : map)
	{
		std::cout << a << b;
	}
```
generates

```
main.cpp:36:20: error: use c++14 style type templates [modernize-type-traits,-warnings-as-errors]
   36 |         for (const auto& [a, b] : map)
      |                           ^
      |                           _t
main.cpp:36:23: error: use c++14 style type templates [modernize-type-traits,-warnings-as-errors]
   36 |         for (const auto& [a, b] : map)
      |                              ^
      |                              _t


```

**Comment by vbvictor** [comment]
https://godbolt.org/z/9c7Gab8oc
This is odd TBH, `type-traits` seems unrelated to structured bindings.

**Comment by vbvictor** [comment]
This bug gave 63 FPs on `clang-tools-extra/clang-tidy` which is… very unfortunate.
Need to look into it.

**Comment by vbvictor** [comment]
This must have broke it https://github.com/llvm/llvm-project/pull/147835.

**Comment by mizvekov** [comment]
Another point here is that the matcher didn't previously handle a NestedNameSpecifier of identifier kind.

Now after [#147835](https://github.com/llvm/llvm-project/pull/147835) these kinds of NestedNameSpecifiers are represented as DependentNameTypes, which this matcher handles.
